routers/web/auth/oauth.go

```diff
@@ -1054,19 +1057,17 @@ func getClaimedGroups(source *oauth2.Source, gothUser *goth.User) container.Set[
 	return claimValueToStringSet(groupClaims)
 }
 
-func setUserAdminAndRestrictedFromGroupClaims(source *oauth2.Source, u *user_model.User, gothUser *goth.User) bool {
+func getUserAdminAndRestrictedFromGroupClaims(source *oauth2.Source, gothUser *goth.User) (isAdmin, isRestricted optional.Option[bool]) {
 	groups := getClaimedGroups(source, gothUser)
 
-	wasAdmin, wasRestricted := u.IsAdmin, u.IsRestricted
-
 	if source.AdminGroup != "" {
-		u.IsAdmin = groups.Contains(source.AdminGroup)
+		isAdmin = optional.Some(groups.Contains(source.AdminGroup))
```

Comment:
Yes, but I mean using `optional.FromNonDefault(groups.Contains(source.AdminGroup))`

Reply:
If I'm not mistaken, that should work as well, right?

Reply:
Ah, I just noticed my mistake: In that case, `false` can never be set.

Reply:
No, if `groups.Contains` returns `false` that's the default for boolean and `FromNonDefault` returns a none optional. Then this field is ignored in the update because it has no value.